ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJBTransformers.java

```diff
@@ -303,7 +305,17 @@ private static void registerTransformers_7_0_0(ResourceTransformationDescription
                 .rejectChildResource(PathElement.pathElement(EJB3SubsystemModel.REMOTE_HTTP_CONNECTION));
     }
 
-
+    /*
+     * Transformers for changes in model version 9.0.0
+     */
+    private static void registerTransformers_8_0_0(ResourceTransformationDescriptionBuilder subsystemBuilder) {
+        // Reject attribute legacy-compliant-principal-propagation
+        subsystemBuilder.addChildResource(PathElement.pathElement(EJB3SubsystemModel.APPLICATION_SECURITY_DOMAIN))
+                .getAttributeBuilder()
+                .setDiscard(DiscardAttributeChecker.UNDEFINED, EJB3SubsystemModel.LEGACY_COMPLIANT_PRINCIPAL_PROPAGATION)
+                .addRejectCheck(RejectAttributeChecker.DEFINED, EJB3SubsystemModel.LEGACY_COMPLIANT_PRINCIPAL_PROPAGATION)
```

Comment:
Correct me if I'm wrong, but it looks like you can also also discard this attribute if set to the default value (true), as this is consistent with the legacy implied behavior.

Reply:
@pferraro yes true. I updated the PR

Reply:
`connectors` and `remote-http-connection` should already be handled by `registerTransformers_7_0_0`, with the chained transformers. Why do we need to duplicate the same in `registerTransformers_8_0_0`?

Reply:
@chengfang That was my initial thought as well, however after I changed it the MixedDomainDeployment730TestCase started to pass for me locally.

Reply:
@chengfang Although looking at the CI now it seems I have messed up my local configuration and tests are still failing. So something else is wrong.

Reply:
@chengfang @pferraro Actually I was looking at the wrong CI result before. Copying contents of registerTransformers_7_0_0 to registerTransformers_8_0_0 did help and all tests passed now.

Reply:
@chengfang please let me know if you agree with this change.

Reply:
let me check with @rachmatowicz , who did the `connectors` refactoring in ejb3 8.0.0 version. Richard, can you check why we need to duplicate the `connectors` part in `registerTransformers_8_0_0()`?

Reply:
We can look into this EJB transformer issue as a separate one without blocking this feature, since this is not core feature of this PR.

Reply:
@chengfang that might be good. But this feature require merging it

Reply:
@chengfang There may have been an issue with the connectors transformers - as it look like there were no EAP 7.3 transformation tests until now.  I agree that this does not need to block this PR, but we should fix this before the WF 23 final release.